bids2table/entities.py

```diff
@@ -4,28 +4,38 @@
 
 import re
 import warnings
-from dataclasses import asdict, dataclass, field, fields
+from dataclasses import asdict, dataclass, field, fields, make_dataclass
 from functools import lru_cache
 from pathlib import Path
 from types import MappingProxyType
-from typing import Any, Callable, Dict, Iterable, List, Optional, Union
+from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple, Union
 
+import bidsschematools.schema
 import pandas as pd
 from elbow.typing import StrOrPath
 from typing_extensions import get_args, get_origin
 
-BIDS_DATATYPES = (
-    "anat",
-    "beh",
-    "dwi",
-    "eeg",
-    "fmap",
-    "func",
-    "ieeg",
-    "meg",
-    "micr",
-    "perf",
-    "pet",
+_bids_schema_path: Optional[str] = None
+
+
+def set_bids_schema_path(path: Optional[str]):
+    """
+    Set the path to the BIDS schema.
+    """
+    global _bids_schema_path
+    _bids_schema_path = path
+
+
+def _get_bids_schema() -> Any:
+    """
+    Get the BIDS schema.
+    """
+    global _bids_schema_path
+    return bidsschematools.schema.load_schema(schema_path=_bids_schema_path)
+
+
+BIDS_DATATYPES = tuple(
+    o.value for o in dict(_get_bids_schema().objects.datatypes).values()
```

Comment:
This is already a dict-like:

```suggestion
    o.value for o in _get_bids_schema().objects.datatypes.values()
```